tests/queries/tests.py

```diff
@@ -2640,9 +2640,15 @@ def setUpTestData(cls):
     def test_infinite_loop(self):
         # If you're not careful, it's possible to introduce infinite loops via
         # default ordering on foreign keys in a cycle. We detect that.
-        with self.assertRaisesMessage(FieldError, 'Infinite loop caused by ordering.'):
+        with self.assertRaisesMessage(
+            FieldError,
+            "Infinite loop caused by ordering (\"((('y_id', 'id'),), (('x_id', 'id'),))\" on queries.LoopY.x)"
+        ):
             list(LoopX.objects.all())  # Force queryset evaluation with list()
-        with self.assertRaisesMessage(FieldError, 'Infinite loop caused by ordering.'):
+        with self.assertRaisesMessage(
+            FieldError,
+            "Infinite loop caused by ordering (\"((('z_id', 'id'),), (('z_id', 'id'),))\" on queries.LoopZ.z)"
+        ):
```

Comment:
Can probably use an intermediary `msg` variable to avoid the weird line wrapping in both cases.